File: services/ai-storage-service/src/services/aiProviders/azureOpenai.js
const fetch = require('node-fetch');
const env = require('../../config/env');

async function chat({ messages }) {
  if (!env.ai.azureEndpoint || !env.ai.azureDeployment) {
    throw Object.assign(new Error('AZURE_OPENAI_ENDPOINT / AZURE_OPENAI_DEPLOYMENT not configured'), { status: 500 });
  }
  const url = `${env.ai.azureEndpoint}/openai/deployments/${env.ai.azureDeployment}/chat/completions?api-version=2024-06-01`;
  const response = await fetch(url, {
    method: 'POST',
    headers: { 'content-type': 'application/json', 'api-key': env.ai.apiKey },
    body: JSON.stringify({ messages }),
  });
  if (!response.ok) {
    const text = await response.text();
    throw Object.assign(new Error(`Azure OpenAI error: ${text}`), { status: 502 });
  }
  const data = await response.json();
  return {
    content: data.choices?.[0]?.message?.content || '',
    usage: data.usage || {},
  };
}

module.exports = { chat };

File: services/ai-storage-service-python/services/ai_providers/azure_openai.py
import httpx
from config.settings import settings
from loguru import logger
from typing import Dict, Any


async def chat(messages: list[Dict[str, str]], model: str = None) -> Dict[str, Any]:
    if not settings.azure_openai_endpoint or not settings.azure_openai_deployment:
        raise Exception("AZURE_OPENAI_ENDPOINT / AZURE_OPENAI_DEPLOYMENT not configured")
    
    url = f"{settings.azure_openai_endpoint}/openai/deployments/{settings.azure_openai_deployment}/chat/completions?api-version=2024-06-01"
    headers = {
        "content-type": "application/json",
        "api-key": settings.ai_api_key
    }
    payload = {"messages": messages}
    
    async with httpx.AsyncClient() as client:
        response = await client.post(url, headers=headers, json=payload)
        
    if not response.is_success:
        error_text = response.text
        logger.error(f"Azure OpenAI error: {error_text}")
        raise Exception(f"Azure OpenAI error: {error_text}")
    
    data = response.json()
    return {
        "content": data.get("choices", [{}])[0].get("message", {}).get("content", ""),
        "usage": data.get("usage", {})
    }
